From 108e74b3e770a1d12eda5ed8dea7ca58d5e90cff Mon Sep 17 00:00:00 2001
From: Alvaro Lopez Garcia <aloga@ifca.unican.es>
Date: Mon, 23 Apr 2012 16:40:38 +0200
Subject: [PATCH] Fixes bug 987335.

Revert bug introduced by commit a837f92e that removed
console_log from get_console_output()

Change-Id: I22a14b5f50c2df0486420b38137328ac87844c1f
---
 nova/virt/libvirt/connection.py |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/nova/virt/libvirt/connection.py b/nova/virt/libvirt/connection.py
index 888be92..eb0649b 100644
--- a/nova/virt/libvirt/connection.py
+++ b/nova/virt/libvirt/connection.py
@@ -1006,6 +1006,7 @@ class LibvirtConnection(driver.ComputeDriver):
 
         self._chown_console_log_for_instance(instance['name'])
         data = self._flush_libvirt_console(pty)
+        console_log = self._get_console_log_path(instance_name)
         fpath = self._append_to_file(data, console_log)
 
         return libvirt_utils.load_file(fpath)
@@ -1147,9 +1148,12 @@ class LibvirtConnection(driver.ComputeDriver):
         libvirt_utils.mkfs('swap', target)
 
     @staticmethod
-    def _chown_console_log_for_instance(instance_name):
-        console_log = os.path.join(FLAGS.instances_path, instance_name,
-                                   'console.log')
+    def _get_console_log_path(instance_name):
+        return os.path.join(FLAGS.instances_path, instance_name,
+                'console.log')
+
+    def _chown_console_log_for_instance(self, instance_name):
+        console_log = self._get_console_log_path(instance_name)
         if os.path.exists(console_log):
             libvirt_utils.chown(console_log, os.getuid())
 
